templates/test/dev/cluster-template-custom-builds.yaml

```diff
@@ -469,12 +470,16 @@ spec:
         permissions: "0744"
       - content: |
           $ErrorActionPreference = 'Stop'
-
+          # Need to add manually AZ to PATH as it is not added without a reset
+          Write-Host "Adding AZ to path:"
+          $env:PATH +=";C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin"
+          echo "running az login via identity"
+          az login --identity
           echo "Use OOT credential provider"
           mkdir C:\var\lib\kubelet\credential-provider
-          curl.exe --retry 10 --retry-delay 5 -L "https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_BLOB_CONTAINER_NAME}/${IMAGE_TAG_ACR_CREDENTIAL_PROVIDER}/azure-acr-credential-provider.exe" --output C:\var\lib\kubelet\credential-provider\acr-credential-provider.exe
+          az storage blob download --blob-url "https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_BLOB_CONTAINER_NAME}/${IMAGE_TAG_ACR_CREDENTIAL_PROVIDER}/azure-acr-credential-provider.exe" --file "C:\var\lib\kubelet\credential-provider\acr-credential-provider.exe" --auth-mode login
```

Comment:
The source of truth for these scripts looks to be in the `oot-credential-provider*.yaml` files here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/5493e53d1aff013b9a093c9732559f12b51b01ca/templates/test/ci/prow-ci-version/patches

Making these changes there then running `make generate` should get the "Verify" CI check green.

Reply:
Ah thanks @nojnhuh. I think there was an attempt to do this more broadly and it didn't work so I am not sure if making this change to those files is the right thing to do?

https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/35dfb57718f2215e5145be07b38f2363c173fb7b

Reply:
@jackfrancis Do you remember exactly what the issue was with #4724? I can't think of any blockers there anymore. I've been doing local testing with custom builds/CI versions and have been needing to make these changes in order to run in the corp tenant. I can open a PR unless you know of some reason that obviously still won't work.